applications/minotari_app_grpc/src/authentication/basic_auth.rs

```diff
@@ -36,8 +41,17 @@ pub struct BasicAuthCredentials {
 }
 
 impl BasicAuthCredentials {
-    pub fn new(user_name: String, password: SafePassword) -> Self {
-        Self { user_name, password }
+    pub fn new(user_name: String, password: SafePassword) -> Result<Self, BasicAuthError> {
+        // Validate the username is well formed
+        if user_name.as_bytes().is_empty() || user_name.as_bytes().len() > MAX_USERNAME_LEN {
+            return Err(BasicAuthError::InvalidUsername);
+        }
+        // Validate the password is well formed
+        let bytes = password.reveal().to_vec();
+        let str_password = Zeroizing::new(String::from_utf8(bytes)?);
+        let _parse_result = PasswordHash::parse(&str_password, Encoding::B64)?;
```

Comment:
The use of the `Zeroizing` wrapper is inconsistent here, as all three of these variables represent the same data.